### DISCUSSION TITLE
How to disable/delete a specific module from TWRP recovery after bootloop?

### DISCUSSION THREAD

**Discussion by MML-Pro** [LSPosed/LSPosed#968] [Q&A] [answered]
My device got a blootloop after installed Android Faker module, I need to know how to remove it from TWRP recovery or disable all module to boot my device again?

**Comment by yujincheng08** [comment, marked as answer]
Remove the module's apk.

**Comment by PeterNjeim** [comment]
Here are the steps I used to edit which modules are enabled from TWRP (computer is required, however you might be able to find ARM binaries for sqlite3, but you'd need a PC to transfer them to your phone anyways, so computer is still required).

1. Download the SQLite3 binary for your operating system: https://www.sqlite.org/download.html ("Precompiled Binaries for Linux/Mac OS X (x86)/Windows". Make sure to select "sqlite-tools" and not "sqlite-dll").
2. Download ADB for your operating system: https://developer.android.com/studio/releases/platform-tools ("Download SDK Platform-Tools for Windows/Mac/Linux"). Accept the terms and click the download button.
3. Connect your phone to your PC with a USB cable.
4. Extract the platform-tools zip file, and open the platform-tools folder it creates and keep opening the folders until you see a lot of .exe files (on Windows), including adb.exe. Open a terminal in that folder, and run `adb devices` to check if your device is detected. It should show your device ID and `recovery` next to it.
5. Run `adb pull /data/adb/lspd/config/modules_config.db` in your terminal. This will copy the file from the phone to the PC (copy, not move).
6. Go back to your file explorer and extract the sqlite zip file you downloaded and open all the folders inside of the sqlite folder until you reach the 3 .exe files (for Windows) ("sqldiff.exe", "sqlite3.exe", and "sqlite3_analyzer.exe"), and open this folder in a new terminal window.
7. In this terminal window, run the sqlite3 binary with `.\sqlite3.exe path\to\modules_config.db`. Note that on macOS and Linux you will use forward slash instead of backslash, and you won't use ".exe" as the file extension. The `path\to\modules_config.db` must be replaced with the actual path to the file, which should be in the platform-tools folder with adb.exe (on Windows) in it.
8. When sqlite3 opens the database, run `SELECT * FROM modules;` to see all the modules and their status. Find the module you want to disable, and check its `mid` number (first number in the row before the "|" symbol. For example: `1|eu.faircode.xlua|/data/app/~~BT03B3zfy_8sS3_Jnk4fXw==/eu.faircode.xlua-09yHPc3_lYEd3VVNv87MDg==/base.apk|1` means it has an `mid` of `1`, since `1` is the first number.
9. The last number in the row is whether or not it is enabled. `1` means enabled and `0` means disabled. In this case, xprivacylua in my example is enabled since it has a `1` at the end.
10. To change this number to a `0`, we must select the row, and edit the last column, which is called `enabled`. Type `UPDATE modules` then press enter, then type `SET enabled = 0` and press enter, then type `WHERE mid = enter mid here;` (in this case I'll do "WHERE mid = 1;") and press enter (make sure to put the semicolon ";" at the end of the last line).
11. You can check if it worked by typing `SELECT * FROM modules;` and checking if it has a `0` at the end: `1|eu.faircode.xlua|/data/app/~~BT03B3zfy_8sS3_Jnk4fXw==/eu.faircode.xlua-09yHPc3_lYEd3VVNv87MDg==/base.apk|0` and it does.
12. Now you can exit the program by typing Ctrl+C.
13. Now the modules_config.db file is modified, use `adb push path\to\modules_config.db /data/adb/lspd/config/` to copy the file to the phone.
14. Reboot and see if it works.

**Reply by xcsnowcity** [reply]
Your answer is so helpful!
However I want to add that you don't need to download sqlite3 again cuz the Platform-Tools itself contain sqlite3 already.

**Reply by AGuyNamedJens** [reply]
This seems to have no effect for me, still getting a bootloop somehow..

**Reply by zgnenes** [reply]
Thanks for your answer. You are saved my data :D
